CWE.Data/DataAccessLayer.cs

```diff
@@ -132,5 +134,122 @@ public async Task<IEnumerable<Request>> GetRequests()
             return await this.dbContext.Requests
                 .ToListAsync();
         }
+
+        /// <summary>
+        /// Checks if the tag name provided exists in the database.
+        /// </summary>
+        /// <param name="tagName">The name of the tag to fetch..</param>
+        /// <returns>A <see cref="Tag"/> depending on if the tag exists in the database.</returns>
+        public async Task<Tag> FetchTagAsync(string tagName)
+        {
+            var tag = await this.dbContext.Tags
+                .Where(x => x.Name == tagName)
+                .SingleOrDefaultAsync();
+
+            if (tag == null)
+            {
+                throw new ArgumentException("The name provided does not match an existing tag in the database.");
+            }
+
+            return tag;
+        }
+
+        /// <summary>
+        /// Creates a tag.
+        /// </summary>
+        /// <param name="name">The name that the tag should have.</param>
+        /// <param name="ownerId">The owner's user ID of the tag.</param>
+        /// <param name="content">The content that the tag holds.</param>
+        /// <returns><see cref="void"/>.</returns>
+        public async Task CreateTagAsync(string name, ulong ownerId, string content)
+        {
+            var tag = await this.dbContext.Tags
+                .Where(x => x.Name == name)
+                .AnyAsync();
+            if (tag)
+            {
+                throw new Exception("A tag with this name already exists.");
+            }
+
+            this.dbContext.Tags.Add(new Tag
+            {
+                Name = name,
+                OwnerId = ownerId,
+                Content = content,
+            });
+            await this.dbContext.SaveChangesAsync();
+        }
+
+        /// <summary>
+        /// Deletes a tag from the database.
+        /// </summary>
+        /// <param name="tagName">The name of the tag to delete.</param>
+        /// <returns><see cref="void"/>.</returns>
+        public async Task DeleteTagAsync(string tagName)
+        {
+            var tag = await this.dbContext.Tags
+                .Where(x => x.Name == tagName)
+                .SingleOrDefaultAsync();
+
+            if (tag == null)
+            {
+                throw new ArgumentException("The tag provided was not found in the database.");
+            }
+
+            this.dbContext.Tags.Remove(tag);
```

Comment:
```suggestion
            this.dbContext.Remove(tag);
```
No need to refer to Tags DbSet.